lib/Radio/Radio.js

```diff
@@ -4,83 +4,101 @@ import PropTypes from 'prop-types';
 import Text from '../Text';
 
 class Radio extends React.Component {
```

Comment:
Why use a class component rather than a functional component with hooks? In my understanding, the benefits are higher in functional components and there are available on React 16, isn't?
Some of the[ benefits of using functional components](https://djoech.medium.com/functional-vs-class-components-in-react-231e3fbd7108) rather than class components:

- **Simplicity**: Functional components are generally easier to read and write than class components. They also require less code to accomplish the same functionality.
- **Improved performance**: Functional components with hooks can have better performance than class components, because they don't have the overhead of creating a class instance.
- **No more "this" binding issues**: Functional components with hooks don't require the use of "this" and "bind", which can make the code more readable and less error-prone.
- **Better reusability**: Functional components with hooks can be more easily reused across different parts of an application, because they don't have the lifecycle methods and state management that are specific to class components.

Reply:
This change could have several benefits, converting the `Radio` component to a functional component using hooks because it's a simpler and more modern way to write React code. The `useState` hook replaces the `state` property of the class component, while the `useCallback` hook replaces the `this.onChange` function of the class component. The props are passed as arguments to the functional component instead of being accessed via `this.props`.

Reply:
I understand that this recommendation involves a significant rewrite of the component, but do you think it would be worth considering?

What do you think @ldiaz-occ ?

Reply:
As @JaviOCC mentions [here](https://github.com/occmundial/occ-atomic/pull/324#discussion_r1164567329), this is the old library and I don't think it's worth it to keep maintaining it, but multiple projects are still using it so it would be better to migrate all the projects still using it to instead use the new [Atomic](https://github.com/occmundial/atomic). Actually one of the main reasons to make a new library was to start using TS and functional components with hooks (and the required effort to do it in this repo was not worth it).